src/Core/Http/HttpClient.cs

```diff
@@ -82,6 +83,19 @@ public static ICredentialProvider DefaultCredentialProvider
             }
         }
 
+        // TODO: Get rid of this. Horrid to have static properties like this especially in a code path that does not look thread safe.

```

Comment:
I actually agree with the comment and left it on purpose, the static properties (and consequently entire IHttpClient) are bad. I actually tried to implement it outside of NuGet client at first, and again by passing the dependencies in, but various HttpClient bits violate LSP left and right and realized the static properties is the only way for this to work w/o rewriting everything that touches HttpClient.